src/types/api.ts

```diff
@@ -153,13 +153,65 @@ export type Shape = RectShape | PointShape;
 /**
  * Selector which identifies a region of the document defined by a shape.
  *
- * TODO:
- *  - Specify the origin used by the shape coordinates (eg. page, document body)
- *  - Specify the coordinate system
+ * # Anchors
+ *
+ * The shape's coordinates may be relative to an _anchor_ element within the
+ * document. For example a page in a PDF or an `<img>` an an HTML document. For
+ * document types where a location can be fully specified by the shape
+ * coordinates alone (such as images, but not PDFs or HTML documents), the
+ * anchor is optional.
+ *
+ * # Coordinate systems
+ *
+ * Shape selectors should be defined using the natural coordinate system for the
+ * anchor element in the document, enabling an annotation made in one viewer to
+ * be resolved to the same location in a different viewer, with different view
+ * settings (zoom, rotation etc.). For common document and anchor types, these
+ * are as follows:
+ *
+ * - For PDFs, PDF user space coordinates (points), with the origin at the
+ *   bottom-left corner of the page.
+ * - For images, pixels with the origin at the top-left
  */
 export type ShapeSelector = {
   type: 'ShapeSelector';
+
   shape: Shape;
+
+  /**
+   * Specifies the element of the document that the shape is relative to.
+   *
+   * This can be omitted in document types such as images, where the coordinates
+   * on their own can specify a unique location in the document.
+   *
+   * Supported values:
+   *
+   * - "page" - The page identified by the annotation's {@link PageSelector}.
+   */
+  anchor?: 'page';
```

Comment:
My rough plan for images in web pages is that there will be a new `MediaSelector` selector which identifies an image by its URL, associated text labels or other attributes. A shape can then specify `anchor: "media"` to specify it is relative to this selector.